Update destination functions - including variable scoping [DOC-1065] (#7512)
segmentio/segment-docs: src/connections/functions/destination-functions.md

> info ""
> Functions' runtime includes a `fetch()` polyfill using a `node-fetch` package. Check out the [node-fetch documentation](https://www.npmjs.com/package/node-fetch){:target="_blank"} for usage examples.

### Variable scoping 

> When declaring settings variables, make sure to declare them in the function handler rather than globally in your Function. This prevents you leaking the settings values across other function instances. 

Reviewer: ```suggestion
When declaring settings variables, declare them in the function handler rather than globally in your function. This prevents you from leaking the settings values across other function instances. 
```

> The handler for Destination functions is event-specific, for example, `onTrack()`, `onIdentify()`, etc.

Reviewer: ```suggestion
The handler for destination functions is event-specific. For example, you might have an `onTrack()`or `onIdentify()` function handler.
```
